### PR TITLE
RDM-6320-Start case creation as Case worker

### PR DESCRIPTION
**Pull request by shrivsksavit**
#JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6320

### Change description ###
Implement Functional Tests for 'CCD Data Store Api :: Start case creation as Case worker' operation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/aat/resources/features/F-049/F-049.feature [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    Given   a user with [an existing case in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [uses a provides authentic credentials without authorized access to the operation in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 403]
+    And     the response has all the details as expected
+
+  @S-233 @Ignore
+  Scenario: must return 404 when no case found for the given Idam user ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: This scenario doesn't look correct.

File: src/aat/resources/features/F-049/F-049.feature [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    Given   a user with [an existing case in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: step is none sense

File: src/aat/resources/features/F-049/F-049.feature [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    When    a request is prepared with appropriate values
+    And     the request [uses a non-existing Idam user ID in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 403]
+    And     the response has all the details as expected
+
+  @S-233
+  Scenario: must return 403 when no case found for the given Idam user ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: Scenario seems to be wrong

File: src/aat/resources/features/F-049/F-049.feature [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 403]
+    And     the response has all the details as expected
+
+  @S-234
+  Scenario: ignore must return 422 when process could not be started
+    Given   a user with [an existing case in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [process could not be started]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: none sense

File: src/aat/resources/features/F-049/F-049.feature [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 422]
+    And     the response has all the details as expected
+
+  @S-511
+  Scenario: must return 404 when no case found for the given Jurisdiction ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: scenario title's wording is wrong

File: src/aat/resources/features/F-049/F-049.feature [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    Given   a user with [an existing case in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [uses a non-existing Jurisdiction ID in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 404]
+    And     the response has all the details as expected
+
+  @S-512 @Ignore
+  Scenario: must return 404 when no case found for the given Case type ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: title is wrong

File: src/aat/resources/features/F-049/F-049.feature [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    Given   a user with [an existing case in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [uses a non-existing Case type ID in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 404]
+    And     the response has all the details as expected
+
+  @S-512
+  Scenario: must return 422 when no case found for the given Case type ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: title is none sense

File: src/aat/resources/features/F-049/F-049.feature [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    Given   a user with [an existing case in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: what is a user with existing case?

File: src/aat/resources/features/F-049/F-049.feature [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    When    a request is prepared with appropriate values
+    And     the request [uses a non-existing Case type ID in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 422]
+    And     the response has all the details as expected
+
+  @S-513 @Ignore
+  Scenario: must return 404 when no case found for the given Case ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: ?

File: src/aat/resources/features/F-049/F-049.feature [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    When    a request is prepared with appropriate values
+    And     the request [a non-existing Case ID in CCD]
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [return 400]
+    And     the response has all the details as expected
+
+
+  @S-514
+  Scenario: must return 404 when no case found for the given Event ID
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: title seems none sense

File: src/aat/resources/features/F-049/S-231.td.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+      "ctid": "AAT",
+      "cid": "1574079355345891",
+      "etid": "CREATE"
+    }
+  },
+  "expectedResponse": {
+    "body" : {
+      "timestamp" : "[[ANY]]",
+      "status" : 403,
+      "error" : "Forbidden",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-231.td.json]: these attributes are already inheritable from the common data.

File: src/aat/resources/features/F-049/S-234.td.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      "cid": "1574079355345891",
+      "etid": "CREATE"
+    }
+  },
+  "expectedResponse": {
+    "body" : {
+      "exception" : "uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException",
+      "timestamp" : "[[ANY]]",
+      "status" : 422,
+      "error" : "Unprocessable Entity",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-234.td.json]: these attributes are already inheritable from the common data.

File: src/aat/resources/features/F-049/S-511.td.json [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      "cid": "1574079355345891",
+      "etid": "CREATE1"
+    }
+  },
+  "expectedResponse": {
+    "body" : {
+      "exception" : "uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException",
+      "timestamp" : "[[ANY]]",
+      "status" : 404,
+      "error" : "Not Found",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-511.td.json]: these attributes are already inheritable from the common data.

**Comment by shrivsksavit** [comment]
as created s fresh PR for the same i.e. 
https://github.com/hmcts/ccd-data-store-api/pull/655

due to that, I am closing this PR

**Pull request closed without merging**